frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -790,6 +800,74 @@ dbg:guard(FileManagerMenu, 'setUpdateItemTable',
         end
     end)
 
+function FileManagerMenu:moveBookMetadata()
+    local DocSettings = require("docsettings")
+    local FileChooser = self.ui.file_chooser
+    local function scanPath()
+        local sys_folders = { -- do not scan sys_folders
+            ["/dev"]  = true,
+            ["/proc"] = true,
+            ["/sys"]  = true,
+        }
+        local books_to_move = {}
+        local dirs = {FileChooser.path}
+        while #dirs ~= 0 do
+            local new_dirs = {}
+            for _, d in ipairs(dirs) do
+                local ok, iter, dir_obj = pcall(lfs.dir, d)
+                if ok then
+                    for f in iter, dir_obj do
+                        local fullpath = "/" .. f
+                        if d ~= "/" then
+                            fullpath = d .. fullpath
+                        end
+                        local attributes = lfs.attributes(fullpath) or {}
+                        if attributes.mode == "directory" and f ~= "." and f ~= ".."
+                                and FileChooser:show_dir(f) and not sys_folders[fullpath] then
+                            table.insert(new_dirs, fullpath)
+                        elseif attributes.mode == "file" and not util.stringStartsWith(f, "._")
+                                and FileChooser:show_file(f) and DocSettings:hasSidecarFile(fullpath)
+                                and lfs.attributes(DocSettings:getSidecarFile(fullpath), "mode") ~= "file" then
+                            table.insert(books_to_move, fullpath)
+                        end
+                    end
+                end
+            end
+            dirs = new_dirs
+        end
+        return books_to_move
+    end
+    UIManager:show(ConfirmBox:new{
+        text = _("Scan current folder and subfolders for book metadata?"),
+        ok_text = _("Scan"),
+        ok_callback = function()
+            local books_to_move = scanPath()
+            local books_to_move_nb = #books_to_move
+            if books_to_move_nb == 0 then
+                local InfoMessage = require("ui/widget/infomessage")
+                UIManager:show(InfoMessage:new{
+                    text = _("No books with metadata out of preferred location found."),
+                })
+            else
+                UIManager:show(ConfirmBox:new{
+                    text = T(N_("1 book with metadata out of preferred location found.",
+                              "%1 books with metadata out of preferred location found.",
+                              books_to_move_nb), books_to_move_nb) ..
+                              _("\nDo you want to move book metadata to preferred location?"),
```

Comment:
Will need feedback from better English writers than us :)
`preferred location` alone feels odd/broken to me, I feel it should read with "your", `your preferred location`.
Also, `out of (your) preferred location` _sounds_ odd , `not in (your) preferred location` _sounds_ better to me.

Reply:
Copying from where it ended up below due to replying to the email:

> Yes, be consistent. Either "move to ... preferred location" or "do you ...
your preferred location"

Reply:
What about "out of" vs. "not in" - or some other wording, in
`%1 books with metadata out of preferred location found.`

Reply:
Does that mean "found in preferred location"?

Reply:
No, precisely the reverse: there are %1 books that have some metadata that are NOT IN / OUT OF your preferred location - and that can be moved there.

Reply:
That'd be "not in" but maybe something like `%1 books with metadata that can be moved (to preferred location)`?